src/util/system.cpp

```diff
@@ -979,6 +1028,215 @@ std::string ArgsManager::GetChainName() const
+        if (*s.begin() == '[' && *s.rbegin() == ']') {
+            // We don't use sections, so we could possibly just write out the rest of the file - but we need to check for unparsable lines, so we just set a flag to ignore settings from here on
+            ModifyRWConfigFile_WriteRemaining(stream_out, settings_to_change, setFound);
+            inside_group = true;
+            key.clear();
+
+            stream_out << linebegin << s << lineend;
+            continue;
+        }
+
+        n = s.find('=');
+        if (n == std::string::npos) {
+            // Bad line; this causes boost to throw an exception when parsing, so we comment out the entire file
+            stream_in.seekg(0, std::ios_base::beg);
+            stream_out.seekp(0, std::ios_base::beg);
+            if (!(stream_in.good() && stream_out.good())) {
+                throw std::ios_base::failure("Failed to rewind (to comment out existing file)");
+            }
+            // First, write out all the settings we intend to set
+            setFound.clear();
+            ModifyRWConfigFile_WriteRemaining(stream_out, settings_to_change, setFound);
+            // We then define a category to ensure new settings get added before the invalid stuff
+            stream_out << "[INVALID]\n";
+            // Then, describe the problem in a comment
+            stream_out << "# Error parsing line " << lineno << ": " << s << "\n";
+            // Finally, dump the rest of the file commented out
+            while (getline_with_eol(stream_in, s)) {
+                stream_out << "#" << s;
+            }
+            return;
+        }
+
+        if (!inside_group) {
+            // We don't support/use groups, so once we're inside key is always null to avoid setting anything
+            n2 = s.find_last_not_of(ws_chars, n - 1);
+            if (n2 == std::string::npos) {
+                n2 = n - 1;
+            } else {
+                ++n2;
+            }
+            key = s.substr(0, n2);
+        }
+        if ((!key.empty()) && (iterCS = settings_to_change.find(key)) != settings_to_change.end() && setFound.find(key) == setFound.end()) {
+            // This is the key we want to change
+            const std::string& val = iterCS->second;
+            setFound.insert(key);
+            ModifyRWConfigFile_SanityCheck(val);
+            if (has_comment) {
+                // Rather than change a commented line, comment it out entirely (the existing comment may relate to the value) and replace it
+                stream_out << key << "=" << val << "\n";
+                linebegin.insert(linebegin.begin(), '#');
+            } else {
+                // Just modify the value in-line otherwise
+                n2 = s.find_first_not_of(ws_chars, n + 1);
+                if (n2 == std::string::npos) {
+                    n2 = n + 1;
+                }
+                s = s.substr(0, n2) + val;
```

Comment:
Nit: Avoid extra allocations by doing:

```
s = s.substr(0, n2);
s += val;
```

Reply:
I think the current style is more readable.

Reply:
I agree. And readability is probably more important than allocation efficiency in this case :-)